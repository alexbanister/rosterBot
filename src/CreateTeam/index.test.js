import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import CreateTeamContainer, { CreateTeam } from './index';
import React from 'react';


describe('CreateTeam', () => {
  it('Container should always match the snapshot', () => {
    const mockStore = configureStore();
    const initialState = {};
    const store = mockStore(initialState);
    const wrapper = shallow(<CreateTeamContainer
      store = {store}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should always match the snapshot', () => {
    const wrapper = shallow(<CreateTeam />);

    expect(wrapper).toMatchSnapshot();
  });

  it('CreateTeam should update state', () => {
    const initialState = {
      teamName: ''
    };
    const expectation = {
      teamName: 'Super Happy Fun Time'
    };
    const wrapper = shallow(<CreateTeam />);

    expect(wrapper.state()).toEqual(initialState);

    const teamName = wrapper.find('input');
    teamName.simulate('change', { target: { value: 'Super Happy Fun Time' }});

    expect(wrapper.state()).toEqual(expectation);
  });
});
