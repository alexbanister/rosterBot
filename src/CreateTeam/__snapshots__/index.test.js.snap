// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTeam Container should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CreateTeam)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "saveTeamName": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "saveTeamName": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CreateTeam should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateTeam />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Create your team
</h3>,
        <form
          onSubmit={[Function]}
>
          <input
                    name="teamName"
                    onChange={[Function]}
                    placeholder="Team Name"
                    type="text"
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Create Team
          </button>
</form>,
      ],
      "className": "createTeam",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create your team",
        },
        "ref": null,
        "rendered": "Create your team",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="teamName"
              onChange={[Function]}
              placeholder="Team Name"
              type="text"
/>,
            <button
              disabled={true}
              type="submit"
>
              Create Team
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "teamName",
              "onChange": [Function],
              "placeholder": "Team Name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Team",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Create Team",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Create your team
</h3>,
          <form
            onSubmit={[Function]}
>
            <input
                        name="teamName"
                        onChange={[Function]}
                        placeholder="Team Name"
                        type="text"
            />
            <button
                        disabled={true}
                        type="submit"
            >
                        Create Team
            </button>
</form>,
        ],
        "className": "createTeam",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create your team",
          },
          "ref": null,
          "rendered": "Create your team",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="teamName"
                onChange={[Function]}
                placeholder="Team Name"
                type="text"
/>,
              <button
                disabled={true}
                type="submit"
>
                Create Team
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "teamName",
                "onChange": [Function],
                "placeholder": "Team Name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create Team",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Create Team",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
